src/System.Buffers/src/Resources/Strings.resx

```diff
@@ -129,4 +129,7 @@
   <data name="event_BucketExhausted" xml:space="preserve">
     <value>Bucket {0}, with {2} maximum buffers, exhausted of all Buffers of size {1} in Pool {3}</value>
   </data>
+  <data name="NotAPooledBufferException" xml:space="preserve">
+    <value>The given buffer is not a Pooled buffer and therefore cannot be returned to the Pool</value>
```

Comment:
Why "Pooled" and "Pool" are upper cased?

Also, I think the error should say that the buffer was not rented from this pool, instead of saying that it is not "pooled". The buffer might be actually pooled from a different (custom) pool.